code/datums/components/hallucinations.dm

```diff
@@ -0,0 +1,537 @@
+	var/list/name_list
+	var/attacker_prob = 10
+	var/max_attackers = 5
+	var/attacker_list = list()
+	Initialize(timeout=30, image_list=null, name_list=null, attacker_prob=10, max_attackers=5)
+		.=..()
+		if(. == COMPONENT_INCOMPATIBLE)
+			return .
+		src.image_list = image_list
+		src.name_list = name_list
+		src.attacker_prob = attacker_prob
+		src.max_attackers = max_attackers
+
+	do_mob_tick(mob, mult)
+		//I know it's kinda gross, but whatever
+		for(var/obj/fake_attacker/fakey in src.attacker_list)
+			if(fakey.disposed)
+				src.attacker_list -= fakey
+		if(length(attacker_list) > src.max_attackers)
+			return
+		if(probmult(attacker_prob))
+			var/obj/fake_attacker/F
+			var/image/halluc
+			if(isnull(image_list)) //if not specified, let's do a 50/50 of critters or humans
+				var/list/possible_clones = new/list()
+				for(var/mob/living/carbon/human/H in mobs)
+					if (H.stat || H.lying || H.dir == NORTH) continue
+					possible_clones += H
+				if(prob(50) && length(possible_clones)) //try for a human fake attacker
+					var/mob/living/carbon/human/clone = null
+					var/clone_weapon = null
+					clone = pick(possible_clones)
+
+					if (clone.l_hand)
+						clone_weapon = clone.l_hand.name
+					else if (clone.r_hand)
+						clone_weapon = clone.r_hand.name
+
+					F = new/obj/fake_attacker(parent_mob.loc, parent_mob)
+
+					F.name = clone.name
+					F.weapon_name = clone_weapon
+					halluc = image(clone,F)
+					parent_mob.client?.images += halluc
+				else //try for a predefined critter fake attacker
+					var/faketype = pick(concrete_typesof(/obj/fake_attacker) - /obj/fake_attacker) //all but the base type
+					F = new faketype(parent_mob.loc, parent_mob)
+
+			else //image list isn't null, so create a fake attacker with that image
+				F = new /obj/fake_attacker(parent_mob.loc, parent_mob)
+				F.name = "attacker"
+				halluc = image(pick(image_list), F)
+				parent_mob.client?.images += halluc
+
+			if(!isnull(name_list))
+				F.name = pick(name_list)
+			src.attacker_list += F
+		..()
+
+	CheckDupeComponent(timeout, image_list, name_list, attacker_prob, max_attackers)
+		if(image_list ~= src.image_list && name_list ~= src.name_list) //this is the same hallucination, just update timeout and prob
+			if(timeout == -1)
+				src.ttl = timeout
+			else if(src.ttl != -1)
+				src.ttl = world.time + timeout SECONDS //reset timeout
+			src.attacker_prob = attacker_prob
+			src.max_attackers = max_attackers
+			return TRUE //no duplicate
+		else
+			return FALSE //create a new hallucination
+
+//#########################################################
+//                 RANDOM IMAGE OVERRIDE
+//#########################################################
+
+/// Random image override - hallucinate an image on a filtered atom in view with prob per life tick, with an option to add as overlay or replace the icon
+/datum/component/hallucination/random_image_override
+	var/list/image_list
+	var/image_prob = 10
+	var/image_time = 20 SECONDS
```

Comment:
same thing as the other one, `SECONDS` used here and in the actual `SPAWN`

Reply:
```suggestion
	var/image_time = 20
```